In [ ]:
from math import sqrt

## Задание 1

Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку. Выберите метод оценки и оцените есть ли стат.значимые различия между конверсиями в двух группах при alpha = 5%. Дайте краткие рекомендации команде .Результаты: 
1) Число юзеров в группах , которые заходили на сайт в период эксперимента: n1 = 15550 и n2 = 15550 . 
2) Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента: n1 = 164 и n2 = 228 3) Конверсии : conv1 = 1.05% conv2 = 1.47% .

Для оценки возьмем Z-критерий.
Формула Z-критерия: $Z = (p1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))$, где $p = (n1 * p1 + n2 * p2) / (n1 + n2)$

Значения:
* p1, p2 - конверсии в группах,
* N1, N2 - количество пользователей в группах,
* n1, n2 - количество пользователей, которые совершили хотя бы одну покупку.

In [1]:
p1 = 164 / 15550
p2 = 228 / 15550

n1 = 164
n2 = 228

N1 = 15550
N2 = 15550

p =  (n1 * p1 + n2 * p2) / (n1 + n2)

Z = (p1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))

#### Итоговое значение

In [2]:
Z

-3.2110930329358367

Критическое начение Z-критерия при α = 5% равно 1.96. В результате полученная величина значительно меньше критического значения (-3.21 < 1.96).
Вывод: Значительные отличия есть и они в пользу выбора новой гипотезы (упрощенной формы заказа).

## Задание 2

Сравниваем метрику конверсия в покупку.Размер выборки - 10000 элементов в каждой группе. 
Какой статистический критерий тут лучше всего подойдёт и почему ?

#### Ответ

Тут, как и в 1 задании, можно использовать Z-критерий, т.к. есть довольно большой размер выборки (10 000 элементов). Также Z-критерий хорошо подходит для сравнения пропорций, позволяя оценить значимость различий между этими пропорциями (тут - конверсиями).

